practical/groovy/analyse_dataset_ilastik.groovy

```diff
@@ -0,0 +1,239 @@
+import loci.common.DataTools
+
+import ij.IJ
+import ij.ImagePlus
+import ij.ImageStack
+import ij.process.ByteProcessor
+import ij.process.ShortProcessor
+import ij.plugin.frame.RoiManager
+import ij.measure.ResultsTable
+
+
+// Setup
+// =====
+
+// OMERO Server details
+HOST = "workshop.openmicroscopy.org"
+PORT = 4064
+
+//  parameters to edit
+dataset_id = 2331
+USERNAME = "username"
+PASSWORD = "password"
+
+
+def connect_to_omero() {
+    "Connect to OMERO"
+
+    credentials = new LoginCredentials()
+    credentials.getServer().setHostname(HOST)
+    credentials.getServer().setPort(PORT)
+    credentials.getUser().setUsername(USERNAME.trim())
+    credentials.getUser().setPassword(PASSWORD.trim())
+    simpleLogger = new SimpleLogger()
+    gateway = new Gateway(simpleLogger)
+    gateway.connect(credentials)
+    return gateway
+
+}
+
+def get_images(gateway, ctx, dataset_id) {
+    "List all images contained in a Dataset"
+
+    browse = gateway.getFacility(BrowseFacility)
+    ids = new ArrayList(1)
+    ids.add(new Long(dataset_id))
+    return browse.getImagesForDatasets(ctx, ids)
+}
+
+def open_image_plus(HOST, USERNAME, PASSWORD, PORT, group_id, image_id) {
+    "Open the image using the Bio-Formats Importer"
+
+    StringBuilder options = new StringBuilder()
+    options.append("location=[OMERO] open=[omero:server=")
+    options.append(HOST)
+    options.append("\nuser=")
+    options.append(USERNAME.trim())
+    options.append("\nport=")
+    options.append(PORT)
+    options.append("\npass=")
+    options.append(PASSWORD.trim())
+    options.append("\ngroupID=")
+    options.append(group_id)
+    options.append("\niid=")
+    options.append(image_id)
+    options.append("] ")
+    options.append("windowless=true view=Hyperstack ")
+    IJ.runPlugIn("loci.plugins.LociImporter", options.toString())
+}
+
+
+def file_exists(name, list) {
+    for (i = 0; i < list.length; i++) {
+        if (name.equals(list[i].toString())) {
+            return true
+        }
+     }
+    return false
+}
+
+def save_rois_to_omero(ctx, image_id, imp) {
+    " Save ROI's back to OMERO"
+    reader = new ROIReader()
+    roi_list = reader.readImageJROIFromSources(image_id, imp)
+    roi_facility = gateway.getFacility(ROIFacility)
+    result = roi_facility.saveROIs(ctx, image_id, exp_id, roi_list)
+
+    roivec = new ArrayList()
+    j = result.iterator()
+    while (j.hasNext()) {
+        roidata = j.next()
+        roi_id = roidata.getId()
+
+        i = roidata.getIterator()
+        while (i.hasNext()) {
+            roi = i.next()
+            shape = roi[0]
+            t = shape.getZ()
+            z = shape.getT()
+            c = shape.getC()
+            shape_id = shape.getId()
+            roivec.add([roi_id, shape_id, z, c, t])
+        }
+    }
+    return roivec
+}
+
+//Beginning of script
+//Choose a directory where to save the .h5 file
+data_dir = IJ.getDirectory("Choose a directory where to save h5 files")
```

Comment:
I was presented with a directory browser dialog. This is fine, but there was no indication what I should choose. I only found out stydying this code.